python/amici/cxxcodeprinter.py

```diff
@@ -20,9 +20,28 @@ def doprint(self, expr: sp.Expr, assign_to: Optional[str] = None) -> str:
             return code
         except TypeError as e:
             raise ValueError(
-                f'Encountered unsupported function in expression "{expr}": '
-                f'{e}!'
-            )
+                f'Encountered unsupported function in expression "{expr}"'
+            ) from e
+
+    def _print_Max(self, expr):
+        from sympy.functions.elementary.miscellaneous import Max
```

Comment:
implement this as a generic function `_print_MinMax` and implement min/max as specialization `_print_MinMax(self, expr, {'min', 'max'})?